lib/fast_jsonapi/object_serializer.rb

```diff
@@ -170,68 +176,61 @@ def has_many(relationship_name, options = {})
-          relationship_type: :has_many,
-          cached: options[:cached] || false,
-          polymorphic: fetch_polymorphic_option(options)
+          is_collection: true,
+          cached: options[:cached] || false
         }
         add_relationship(name, relationship)
       end
 
-      def belongs_to(relationship_name, options = {})
-        name = relationship_name.to_sym
-        serializer_key = options[:serializer] || relationship_name.to_sym
-        key = options[:key] || run_key_transform(relationship_name)
-        record_type = options[:record_type] || run_key_transform(relationship_name)
-        add_relationship(name, {
-          key: key,
-          name: name,
-          id_method_name: options[:id_method_name] || (relationship_name.to_s + '_id').to_sym,
-          record_type: record_type,
-          object_method_name: options[:object_method_name] || name,
-          serializer: compute_serializer_name(serializer_key),
-          relationship_type: :belongs_to,
-          cached: options[:cached] || true,
-          polymorphic: fetch_polymorphic_option(options)
-        })
-      end
-
       def has_one(relationship_name, options = {})
         name = relationship_name.to_sym
         serializer_key = options[:serializer] || name
         key = options[:key] || run_key_transform(relationship_name)
-        record_type = options[:record_type] || run_key_transform(relationship_name)
         add_relationship(name, {
           key: key,
           name: name,
           id_method_name: options[:id_method_name] || (relationship_name.to_s + '_id').to_sym,
-          record_type: record_type,
+          record_type: fetch_record_type_option(options),
           object_method_name: options[:object_method_name] || name,
           serializer: compute_serializer_name(serializer_key),
-          relationship_type: :has_one,
-          cached: options[:cached] || false,
-          polymorphic: fetch_polymorphic_option(options)
+          is_collection: false,
+          cached: options[:cached] || false
         })
       end
 
+      alias belongs_to has_one
+
       def compute_serializer_name(serializer_key)
         namespace = self.name.gsub(/()?\w+Serializer$/, '')
         serializer_name = serializer_key.to_s.classify + 'Serializer'
         return (namespace + serializer_name).to_sym if namespace.present?
         (serializer_key.to_s.classify + 'Serializer').to_sym
       end
 
-      def fetch_polymorphic_option(options)
-        option = options[:polymorphic]
-        return false unless option.present?
+      def fetch_record_type_option(options)
+        option = options[:record_type]
+        # Return it right away if it's a symbol or a Hash:
+        return option if option.is_a?(Symbol) || option.respond_to?(:keys)
+        # Return an empty Hash so the autodetect + memoization kicks in
+        {}
+      end
+
+      def fetch_record_type_option(options)
```

Comment:
This method an the one above seem to have the same method signature. May be you meant to delete one?

Reply:
Indeed... last-minute copy & paste error... fixed!